src/main/scala/com/gyoho/adget/utils/HttpClient.scala

```diff
@@ -0,0 +1,42 @@
+package com.gyoho.adget.utils
+
+import java.net.URL
+
+import com.twitter.finagle.{Http, Service, http}
+import com.twitter.util.{Future => TwitterFuture}
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future, Promise, ExecutionContext}
+
+// use URL instead of String b/c I don't want to deal with edge cases handling
+trait HttpClient[T] {
+  def httpGet(url: URL, timeoutInMillis: Long)(implicit ec: ExecutionContext): T
+  def httpPost(url: URL, payload: String, timeoutInMillis: Long)(implicit ec: ExecutionContext): T
+}
+
+object ScalaHttpClient extends HttpClient[String] {
+  @throws(classOf[java.io.IOException])
+  override def httpGet(url: URL, timeoutInMillis: Long)(implicit ec: ExecutionContext) : String = {
```

Comment:
Try to use akka http. Also, we shouldn't block here.  Return a Future so that the result is asynchronous.